src/modules/sync/advancedDiff.ts

```diff
@@ -0,0 +1,543 @@
+  ],
+  [
+    "{{num_types_removed}}",
+    ({ contentTypes }: DiffData) => contentTypes.deleted.size.toString(),
+  ],
+  [
+    "{{num_types_updated}}",
+    ({ contentTypes }: DiffData) =>
+      [...contentTypes.updated]
+        .filter(([, v]) => v.length > 0)
+        .length.toString(),
+  ],
+  [
+    "{{num_snippets_added}}",
+    ({ contentTypeSnippets }: DiffData) => contentTypeSnippets.added.length.toString(),
+  ],
+  [
+    "{{num_snippets_removed}}",
+    ({ contentTypeSnippets }: DiffData) => contentTypeSnippets.deleted.size.toString(),
+  ],
+  [
+    "{{num_snippets_updated}}",
+    ({ contentTypeSnippets }: DiffData) =>
+      [...contentTypeSnippets.updated]
+        .filter(([, v]) => v.length > 0)
+        .length.toString(),
+  ],
+  [
+    "{{num_taxonomy_added}}",
+    ({ taxonomyGroups }: DiffData) => taxonomyGroups.added.length.toString(),
+  ],
+  [
+    "{{num_taxonomy_removed}}",
+    ({ taxonomyGroups }: DiffData) => taxonomyGroups.deleted.size.toString(),
+  ],
+  [
+    "{{num_taxonomy_updated}}",
+    ({ taxonomyGroups }: DiffData) =>
+      [...taxonomyGroups.updated]
+        .filter(([, v]) => v.length > 0)
+        .length.toString(),
+  ],
+  [
+    "{{source_env_id}}",
+    ({ sourceEnvironmentId, folderName }: DiffData) => sourceEnvironmentId ?? folderName!,
+  ],
+  [
+    "{{target_env_id}}",
+    ({ targetEnvironmentId }: DiffData) => targetEnvironmentId,
+  ],
+  ["{{datetime_generated}}", () => new Date().toUTCString()],
+  [
+    "{{env_link_disabler}}",
+    ({ folderName }: DiffData) => (folderName ? "disabled" : ""),
+  ],
+  [
+    "{{hide_empty_updated_snippets}}",
+    ({ contentTypeSnippets }: DiffData) =>
+      [...contentTypeSnippets.updated].filter(([, v]) => v.length > 0)
+          .length === 0
+        ? "hidden"
+        : "",
+  ],
+]);
+
+const patchOpRendererMap: ReadonlyMap<
+  Operation,
+  RenderFunction<PatchOperation>
+> = new Map<Operation, RenderFunction<any>>([
+  ["addInto", renderOp(renderAddIntoOpDetail)],
+  ["move", renderOp(renderMoveOpDetail)],
+  ["remove", renderOp(renderRemoveOpDetail)],
+  ["replace", renderOp(renderReplaceOpDetail)],
+]);
+
+const modifierMap = new Map<Operation, string>([
+  ["addInto", `<span class="num-added modifier-icon">+</span>`],
+  ["move", `<span class="num-modified modifier-icon">⤷</span>`],
+  ["remove", `<span class="num-removed modifier-icon">−</span>`],
+  ["replace", `<span class="num-modified modifier-icon">⇄</span>`],
+]);
+
+const replaceEntityPathRenderers: EntityPathRenderer[] = [
+  {
+    regex: /^\/name$/,
+    render: () => `Content type name`,
+  },
+  {
+    regex: /^\/elements\/codename:([^/]+)\/([^/]+)$/,
+    render: (match: string[]) => `Property <strong>${match[2]}</strong> of element <strong>${match[1]}</strong>`,
+  },
+  {
+    regex: /^\/content_groups\/codename:([^/]+)\/name$/,
+    render: (match: string[]) => `Content group <strong>${match[1]}</strong> name`,
```

Comment:
Nitpick(something to think about): Content group with codename added? I tried diffing environment and there is difference between name and codename :D also I cannot see the name of the content group added - but I don't care about the name, but maybe somebody will? What do you think? :D

Reply:
`WON'T DO` CO S TYM UROBIS TY DEGESKO?